src/js/device-state.js

```diff
@@ -0,0 +1,94 @@
+import on from './on';
+import { add } from './class-list';
+import ownerWindow from './owner-window';
+import throttle from './throttle';
+
+const deviceStateClass = '.js-device-state';
+const regexWhiteSpace = /\s/g;
+const regexUnquote = /^['"]+|['"]+$/g;
+
+export default class DeviceStateObserver {
+  constructor() {
+    this.window = null;
+    this.lastContent = '';
+    this.hasStateChanged = false;
+  }
+
+  init() {
+    if (!this.node) {
+      this.node = document.body;
+      add(this.node, deviceStateClass);
+    }
+
+    const _handleResize = throttle(this.handleResize, 100);
+    this.offResize = on(ownerWindow(this.node), 'resize', _handleResize);
```

Comment:
result of call to `ownerWindow` could be cashed

Reply:
thanks. i leave it for the moment (time issue)